cms/models/pagemodel.py

```diff
@@ -355,7 +355,13 @@ def save(self, no_signals=False, change_state=True, commit=True,
         from cms.utils.permissions import _thread_locals
         user = getattr(_thread_locals, "user", None)
         if user:
-            self.changed_by = user.username
+            if hasattr(user, 'get_full_name') and user.get_full_name():
+                username = user.get_full_name()
+            elif hasattr(user, 'email') and user.email.strip():
```

Comment:
I am paranoid about `None` sneaking in where you expect a blank string, so I'd do `elif getsattr(user, 'email', None) and user.email.strip()`.